src/util/std_types.h

```diff
@@ -1671,4 +1671,66 @@ inline complex_typet &to_complex_type(typet &type)
   return static_cast<complex_typet &>(type);
 }
 
+class function_typet:public typet
```

Comment:
What about `pure_function_typet` to reduce ambiguity?

Reply:
It's not a pure function -- pure functions are still a programming language concept, whereas this is a function as in mathematics. Perhaps mathematical_function_typet? Just getting long.

Reply:
A long _descriptive_ name might still be better than one that risks to cause confusion. We have "goto_functions", to really no one can assume that we are strict about "function".

Reply:
Ok, will go for mathematical_function, as it's otherwise too hard not to think about a function in a program.